src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -342,6 +355,21 @@ public override BoundNode VisitLambda(BoundLambda node)
         {
             if (_inExpressionLambda)
             {
+                var lambda = node.ExpressionSymbol as MethodSymbol;
+                if ((object)lambda != null)
```

Comment:
I think you can just say `node.Symbol`, which is a LambdaSymbol, which inherits from MethodSymbol (i.e., I don't think this check can ever fail).

Reply:
Sounds good